File: Programmers/Lv.1/성격 유형 검사하기/118666.java
import java.util.*;

class Solution {
    public String solution(String[] survey, int[] choices) {
        String answer = "";
        
        String[] arr = {"R", "T", "C", "F", "J", "M", "A", "N"};
        Map<String, Integer> map = new HashMap<>();
        for (String a : arr) map.put(a, 0);
        
        for (int i = 0; i < survey.length; i++) {
            String[] s = survey[i].split("");
            int c = choices[i];
            
            if (c > 4) {
                map.put(s[1], map.get(s[1]) + (c - 4));
            } else {
                map.put(s[0], map.get(s[0]) + (4 - c));
            }
        }
        
        answer += (map.get("R") >= map.get("T")) ? "R" : "T";
        answer += (map.get("C") >= map.get("F")) ? "C" : "F";
        answer += (map.get("J") >= map.get("M")) ? "J" : "M";
        answer += (map.get("A") >= map.get("N")) ? "A" : "N";
        
        return answer;
    }
}

File: Programmers/Lv.1/성격 유형 검사하기/118666.js
function solution(survey, choices) {
    let answer = '';
    
    const obj = {
        'R' : 0, 'T' : 0,
        'C' : 0, 'F' : 0,
        'J' : 0, 'M' : 0,
        'A' : 0, 'N' : 0
    };
    
    survey.forEach((v, i) => {
       const c = choices[i];
        
        if (c > 4) {
            obj[v.charAt(1)] += c - 4;
        } else {
            obj[v.charAt(0)] += 4 - c;
        }
    });
    
    answer += (obj['R'] >= obj['T']) ? 'R' : 'T';
    answer += (obj['C'] >= obj['F']) ? 'C' : 'F';
    answer += (obj['J'] >= obj['M']) ? 'J' : 'M';
    answer += (obj['A'] >= obj['N']) ? 'A' : 'N';
    
    return answer;
}
